### PR TITLE
docs: Normalize links to https or their web archived version when unavailable

### PR DESCRIPTION
**Pull request by echoix**
I had worked on last February to change http links to https when possible. Down that rabbit hole, I opened every link I saw to see if it had a https version, and when the link was dead, I made a separate research to find where that content is found now. When it was impossible to find an equivalent content hosted somewhere, I changed the link to a link to the wayback machine. Since I used the Internet Archive extension at that time, I also made sure that every link I opened was at least backed up in the last 90 days, as of February 2024. I also used the wayback machine to help find the location of a moved website that didn't have redirects anymore. 

I was extensively using search and replace, and also normalized multiple different URLs from different files to be uniformized across the repo, when it made sense. Example: taking permalinks when available, using the recommended URL scheme for publication ids, DOI, etc.

I had an open PR in my fork of this work where I asked for some feedback before posting here.

I'm opening the PR right away, but I'm currently rebasing the commits to remove already committed fixes (to enable linting), and some URLs recently independently updated.

### CODE REVIEW

File: src/general/g.proj.all/g.proj.all.html [hunk 1]
```diff
@@ -24,5 +24,5 @@ <h2>SEE ALSO</h2>
 
 <h2>AUTHORS</h2>
 
-Anna Petrasova, <a href="http://gis.ncsu.edu/osgeorel/">NCSU OSGeoREL</a>,<br>
-Vaclav Petras, <a href="http://gis.ncsu.edu/osgeorel/">NCSU OSGeoREL</a>
+Anna Petrasova, <a href="https://web.archive.org/web/20130707111603/https://gis.ncsu.edu/osgeorel/">NCSU OSGeoREL</a>,<br>
+Vaclav Petras, <a href="https://web.archive.org/web/20130707111603/https://gis.ncsu.edu/osgeorel/">NCSU OSGeoREL</a>
```
Comment (neteler) [inline src/general/g.proj.all/g.proj.all.html]: Probably point to `https://geospatial.ncsu.edu/geoforall/">NCSU GeoForAll Lab</a>`?

File: src/gui/wxpython/wx.metadata/g.gui.cswbrowser/g.gui.cswbrowser.html [hunk 2]
```diff
@@ -86,4 +86,4 @@ <h2>AUTHORS</h2>
 Matej Krejci, <a href="https://geo.fsv.cvut.cz/gwiki/osgeorel">OSGeoREL</a>
 at the Czech Technical University in Prague, developed
 during <a href="https://trac.osgeo.org/grass/wiki/GSoC/2014/MetadataForGRASS">Google
-    Summer of Code 2015</a> (mentor: Martin Landa)
+    Summer of Code 2015</a> (mentors: Martin Landa)
```
Comment (neteler) [inline src/gui/wxpython/wx.metadata/g.gui.cswbrowser/g.gui.cswbrowser.html]:
```suggestion
    Summer of Code 2015</a> (mentor: Martin Landa)
```

File: src/imagery/i.landsat8.swlst/helpers.py [hunk 3]
```diff
@@ -95,10 +95,10 @@ def mask_clouds(qa_band, qa_pixel):
     - support for multiple qa_pixel values (eg. input as a list of values)
 
     Create and apply a cloud mask based on the Quality Assessment Band
-    (BQA.) Source: <http://landsat.usgs.gov/L8QualityAssessmentBand.php
+    (BQA.) Source: <https://landsat.usgs.gov/L8QualityAssessmentBand.php
 
     See also:
-    http://courses.neteler.org/processing-landsat8-data-in-grass-gis-7/#Applying_the_Landsat_8_Quality_Assessment_%28QA%29_Band
+    https://courses.neteler.org/processing-landsat8-data-in-grass-gis-7/#Applying_the_Landsat_8_Quality_Assessment_%28QA%29_Band
```
Comment (neteler) [inline src/imagery/i.landsat8.swlst/helpers.py]:
```suggestion
    https://neteler.org/blog/processing-landsat8-data-in-grass-gis-7/#Applying_the_Landsat_8_Quality_Assessment_%28QA%29_Band
```

File: src/imagery/i.landsat8.swlst/i.landsat8.swlst.html [hunk 4]
```diff
@@ -244,15 +244,15 @@ <h3 id="example-figures">Example figures</h3>
 </div>
 <h2 id="todo">TODO</h2>
 <ul>
-<li>Go through <a href="http://trac.osgeo.org/grass/wiki/Submitting/Python">Submitting Python</a></li>
+<li>Go through <a href="https://trac.osgeo.org/grass/wiki/Submitting/Python">Submitting Python</a></li>
```
Comment (neteler) [inline src/imagery/i.landsat8.swlst/i.landsat8.swlst.html]:
```suggestion
<li>Go through <a href="https://github.com/OSGeo/grass/blob/main/doc/development/style_guide.md#python">Submitting Python</a></li>
```

File: src/raster/r.connectivity/r.connectivity.distance/r.connectivity.distance.py [hunk 5]
```diff
@@ -406,9 +406,8 @@ def main():
     # cost_ewres = reg['ewres']
 
     # Rasterize patches
-    # http://www.gdal.org/gdal_tutorial.html
-    # http://geoinformaticstutorial.blogspot.no/2012/11/convert-
-    # shapefile-to-raster-with-gdal.html
+    # https://web.archive.org/web/20180707051311/http://gdal.org/gdal_tutorial.html
```
Comment (neteler) [inline src/raster/r.connectivity/r.connectivity.distance/r.connectivity.distance.py]:
```suggestion
    # https://gdal.org/en/latest/tutorials/index.html
```

**Review by neteler — APPROVED** [review]
With the few suggested modifications applied good to go.

File: src/raster/r.connectivity/r.connectivity.distance/r.connectivity.distance.py (hunk 5 above)
Comment (echoix) [reply src/raster/r.connectivity/r.connectivity.distance/r.connectivity.distance.py]: Which of the tutorials specifically? I went back to that date (content similar to when committed) as it was more some content on rasters, and wasn't split up yet. Can you infer what tutorial is meant to show, what info is linked?

File: src/general/g.proj.all/g.proj.all.html (hunk 1 above)
Comment (echoix) [reply src/general/g.proj.all/g.proj.all.html]: Yep, following https://github.com/OSGeo/grass-addons/pull/1234#discussion_r1824866399

File: src/raster/r.connectivity/r.connectivity.distance/r.connectivity.distance.py (hunk 5 above)
Comment (neteler) [reply src/raster/r.connectivity/r.connectivity.distance/r.connectivity.distance.py]: I thought that in the GDAL case pointing to an outdated tutorial will not be of much help for users.
But feel free to ignore this suggestion.

This one looks close: https://gdal.org/en/stable/tutorials/raster_api_tut.html

Comment (echoix) [reply src/raster/r.connectivity/r.connectivity.distance/r.connectivity.distance.py]: Ok, we could use that, but I don't see, even historically what pattern it referred to (maybe I didn't go far enough).

Comment (echoix) [reply src/raster/r.connectivity/r.connectivity.distance/r.connectivity.distance.py]:
```suggestion
    # https://gdal.org/en/stable/tutorials/raster_api_tut.html
```

File: src/general/g.proj.all/g.proj.all.html (hunk 1 above)
Comment (echoix) [reply src/general/g.proj.all/g.proj.all.html]:
```suggestion
Anna Petrasova, <a href="https://geospatial.ncsu.edu/geoforall/">NCSU GeoForAll Lab</a>,<br>
Vaclav Petras, <a href="https://geospatial.ncsu.edu/geoforall/">NCSU GeoForAll Lab</a>
```

**Pull request merged**